core/embed/projects/prodtest/cmd/prodtest_secrets.c

```diff
@@ -160,31 +167,54 @@ static void prodtest_secrets_get_mcu_device_key(cli_t* cli) {
   cli_ok_hexdata(cli, output, sizeof(output));
 
 cleanup:
+  memzero(seed, sizeof(seed));
   memzero(mcu_private, sizeof(mcu_private));
 }
 
 #ifndef TREZOR_EMULATOR
 static bool check_device_cert_chain(cli_t* cli, const uint8_t* chain,
                                     size_t chain_size) {
-  ed25519_secret_key mcu_private = {0};
-  if (secret_key_mcu_device_auth(mcu_private) != sectrue) {
+  bool ret = false;
+
+  uint8_t seed[MLDSA_SEEDBYTES] = {0};
+  if (secret_key_mcu_device_auth(seed) != sectrue) {
     cli_error(cli, CLI_ERROR, "`secret_key_mcu_device_auth()` failed.");
-    return false;
+    goto cleanup;
+  }
+
+  uint8_t mcu_public[CRYPTO_PUBLICKEYBYTES] = {0};
+  uint8_t mcu_private[CRYPTO_SECRETKEYBYTES] = {0};
+  if (crypto_sign_keypair_internal(mcu_public, mcu_private, seed) != 0) {
+    cli_error(cli, CLI_ERROR, "`crypto_sign_keypair_internal()` failed.");
+    goto cleanup;
   }
 
+  uint8_t rnd[MLDSA_RNDBYTES] = {0};
+  random_buffer(rnd, sizeof(rnd));
+
   // The challenge is intentionally constant zero.
   uint8_t challenge[CHALLENGE_SIZE] = {0};
-  ed25519_signature signature = {0};
-  ed25519_sign(challenge, sizeof(challenge), mcu_private, signature);
-  memzero(mcu_private, sizeof(mcu_private));
+  uint8_t signature[CRYPTO_BYTES] = {0};
+  size_t siglen = 0;
+  if (crypto_sign_signature_internal(signature, &siglen, challenge,
```

Comment:
Why do you use `crypto_sign_signature_internal` and not `crypto_sign_signature`?

Reply:
The `pre` parameter (pointer to prefix string) `(const uint8_t*)"\0\0"` represents the encoding of an empty context string and the 2 its length. If FIPS 204, Algortithm 2, line 10:
> 𝑀 ′ ← BytesToBits(IntegerToBytes(0, 1) ∥ IntegerToBytes(|𝑐𝑡𝑥|, 1) ∥ 𝑐𝑡𝑥) ∥ 𝑀

It corresponds to the part BytesToBits(IntegerToBytes(0, 1) ∥ IntegerToBytes(|𝑐𝑡𝑥|, 1) ∥ 𝑐𝑡𝑥).